rust/src/storage/file/mod.rs

```diff
@@ -102,7 +102,8 @@ impl StorageBackend for FileStorageBackend {
             fs::create_dir_all(parent).await?;
         }
         let mut f = fs::OpenOptions::new()
-            .create_new(true)
+            .create(true)
+            .truncate(true)
             .write(true)
             .open(path)
             .await?;
```

Comment:
This was failing with already exists for fs because of `_last_checkpoint` overwrite. I was thinking about creating new function like `replace_obj` but then, in out s3 backend, the `put_obj` replaces exists by default and there's no workaround for it. So instead, I've updated this to be the same semantically as in s3

cc @houqp @xianwill

Reply:
I think your change introduces the right behavior, only `rename_obj` is supposed to check for already exists error.

Reply:
Sure, I'll add an issue.